pluginExamples/OEM/Worek.py

```diff
@@ -48,35 +30,174 @@
 activity = (sample mE - blank mE) / 10.6
 AChE = activity * 1.58e3
 activity = (sample mE - blank mE) * 1.58e3 / 10.6 
+
+=== PLUGINS IN ENLIGHTEN ===
+
+- get_intensity_from_wavelength
+- get_widget_from_name
+
 """
 
+def getY(x, domain, codomain):
+    """
+    given two arrays (domain, codomain) which describe a collection of points
+      domain: x0 x1 x2 x3 x4 ... xn
+    codomain: y0 y1 y2 y3 y4 ... yn
+
+    return the element from codomain such that the corresponding element from domain = x
+    ex: getY(4, [0,2,4,6,8], [1,3,5,7,9]) --> 5
+
+    if x is not in the domain, return the closest boundary element
+    """
+
+    for i in range(len(domain)-1):
+        if domain[i] <= x < domain[i+1]:
+            return codomain[i]
+
+    if domain[-1] >= x:
```

Comment:
minor optimization note: you can check for endpoints (less than first, greater than last) before the loop.